### ISSUE TITLE
Generation of a code quality report in a JSON file that implements a subset of the Code Climate spec.

### ISSUE THREAD

**Issue by hlemos-4c** [forcedotcom/sfdx-scanner#329]
Hi,

I am trying to make use of the code climate report in Gitlab - a feature that allows to compare the code quality between branches in a merge request, highlighting for example if there was degradation of code quality between the source branch and the destination branch. But unfortunately the report format generated by Salesforce CLI Scanner doesn't meet the Code Climate Specification: https://docs.gitlab.com/ee/user/project/merge_requests/code_quality.html#implementing-a-custom-tool

Having an option in CLI Scanner to generate a report compliant with the code climate specification will add a very useful use case for CLI Scanner when working with gitlab. 

Thank you.

Kind regards,
Hugo

**Comment by jbartolotta-sfdc** [comment]
@hlemos-4c thanks for letting us know about your use case. There are lots of CI/CD formats, we will need to evaluate if this format is popular enough to add as a default option.

In the meantime, I have attached an example JavaScript file that you can use to convert the output of the `sfdx scanner:run --format json` command into the Code Climate format. It would require your CI/CD system to have Node.js installed.

An example invocation would be:
```bash
sfdx scanner:run --target test/code-fixtures/projects/js --format json | node code-climate-transform.js
```
(you will need to remove the `.txt` extension)
[code-climate-transform.js.txt](https://github.com/forcedotcom/sfdx-scanner/files/5846344/code-climate-transform.js.txt)

Let us know if this helps. If your CI/CD system doesn't  have Node.js available, but you do have Python or another language available, I can code something else up.

Thanks.

**Comment by hlemos-4c** [comment]
Hi Jeff,

Thank you for replying to the comment and for providing the JS file to convert the report from one format to the other. I tried and it works well! We will start using CLI scanner in our projects. 

One thing to have in mind for those that may stumble in this issue, I just had to print the output into a file
sfdx scanner:run --target test/code-fixtures/projects/js --format json | node code-climate-transform.js **> code-quality-report.json**

Thank you again Jeff, great work.
